### ISSUE TITLE
Regex search and replace does not work in editor

### ISSUE THREAD

**Issue by lprot** [doublecmd/doublecmd#1748]
### Version

Double Commander
Version: 1.2.0 alpha
Revision: 11393
Commit: e650495a5
Build date: 2024/07/06
Lazarus: 3.99.0.0
FPC: 3.2.2
Platform: x86_64-Darwin-cocoa
OS version: Mac OS X 12.7.5

### Description

File with carriage returns \n (0xA) [A_installer.txt](https://github.com/user-attachments/files/16201894/A_installer.txt)
![image](https://github.com/user-attachments/assets/f0cd7050-fde9-468d-aca1-19c81e098b30)
![image](https://github.com/user-attachments/assets/8a8563db-96e2-4aed-becf-6940c88d67b8)

**Comment by alexx2000** [comment]
You should to enable "Multiline pattern" in this case.

**Comment by lprot** [comment]
Works, so it does not work for the first line because of lazarus bug?

**Comment by alexx2000** [comment]
No problem with the first line. It works wrong with the last line (replaces a whole text).

**Comment by lprot** [comment]
> No problem with the first line. It works wrong with the last line (replaces a whole text).

Then it is a second bug. Because I expect that when you do not enable multiline optione it has to find the first match at least.

**Comment by alexx2000** [comment]
> Then it is a second bug. Because I expect that when you do not enable multiline option it has to find the first match at least.

No, text with '\n' is multiline. Without this option, each line is treated as a separate entity (which does not contain '\n').

> This is very strange that in editor to find a regex you need to enable multiline and in viewer the search just works if you enable regex.

Viewer works on completely different principles. It does not split text into lines.

**Comment by lprot** [comment]
> No, text with '\n' is a multiline text. Without this option, each line is treated as a separate entity (which does not contain '\n').

I do not get this logic. I want to find 0xA or \n in the first line. I enable regex and search for \x0A or \n and find nothing. As soon as I enable multiline then it start working. Then it makes sense always enable multiline. In cases you don't need it, you switch it off manually. Currently it is very confusing.

**Comment by magorium** [comment]
@lprot 
I don't know what your exact aim is here but you can search for the end of a line in a regular expression with "$" (dollar sign).

If you want to change line-ending themselves then the option edit, end of line (then save file) can be used and for about anything else you wish to do with line-endings the multi-line option is required.

**Comment by Alexey-T** [comment]
FPC git 'main' has updated TRegExpr. but not the last TRegExpr. the LAST one is here:
https://github.com/andgineer/TRegExpr
